### DISCUSSION TITLE
How to add a custom REST endpoint at the root?

### DISCUSSION THREAD

**Discussion by blevine** [keycloak/keycloak#31801] [Q&A]
Hi,

I'd like to write a REST API extension that implements an endpoint that lives outside of any realm. So the path would be /admin/my-endpoint rather than having to do /admin/realms/{realmname}/my-endpoint. Is this possible?

**Comment by lamoboos223** [comment]
may I ask what is the logic of the endpoint? because what kind of logic to be applied out side the realm when the realm is the big umbrella of keycloak's architecture.

@blevine

**Reply by blevine** [reply]
We are adding some custom information to Keycloak that is independent of any realm.

**Reply by lamoboos223** [reply]
Unfortunately this is the only example i found by extending keycloak via SPI and it contains realms in the url: https://github.com/dasniko/keycloak-extensions-demo/tree/main/rest-endpoint/src/main/java/dasniko/keycloak/resource

i’m not sure if there is other way to register a web resource that isn’t based on a realm, if you really need that url without realm, ig you can modify the core code and inject your endpoint there.
